packages/api/test/e2e/staking.e2e.ts

```diff
@@ -0,0 +1,302 @@
+      const controllerAddress = keyring.encodeAddress(controllerOpt.unwrap());
+      if (controllerAddress === newController.address) {
+        done();
+      }
+    });
+
+    await api.tx.staking.setController(newController.address).signAndSend(stash);
+  });
+
+});
+
+describe('Staking Governance (Sudo Required)', () => {
+
+  afterAll(async done => {
+    // Ensure era forcing is disabled
+    await api.tx.sudo.sudo(api.tx.staking.forceNewEra())
+      .signAndSend(alice, ({ status }) => status.isInBlock ? done() : null);
+  });
+
+  test('Set target validator count', async done => {
+    const validatorCount = 15;
+    const setValidatorTx = api.tx.staking.setValidatorCount(validatorCount);
+
+    await api.tx.sudo.sudo(setValidatorTx).signAndSend(alice, async ({ status }) => {
+      if (status.isInBlock) {
+        expect((await api.query.staking.validatorCount()).toString()).toEqual(validatorCount.toString());
+        done();
+      }
+    });
+  });
+
+  test('Set minimum bond', async done => {
+    const minimumBond = 1_234;
+    const setMinimumBondTx = api.tx.staking.setMinimumBond(minimumBond);
+
+    await api.tx.sudo.sudo(setMinimumBondTx).signAndSend(alice, async ({ status }) => {
+      if (status.isInBlock) {
+        expect((await api.query.staking.minimumBond()).toString()).toEqual(minimumBond.toString());
+        done();
+      }
+    });
+  });
+
+
+  test('Set invulnerable validators', async done => {
+    const invulnerables: AccountId[] = keyring.getPairs().map(p => p.publicKey as AccountId);
+    const setInvulnerablesTx = api.tx.staking.setInvulnerables(invulnerables);
+
+    await api.tx.sudo.sudo(setInvulnerablesTx).signAndSend(alice, async ({ status }) => {
+      if (status.isInBlock) {
+        expect(
+          (await api.query.staking.invulnerables()).map(k => keyring.encodeAddress(k))
+        ).toEqual(
+          invulnerables.map(k => keyring.encodeAddress(k))
+        );
+        done();
+      }
+    });
+  });
+
+  test('Force no eras', async done => {
+    await api.query.staking.forceEra(
+      (forcing: Forcing) => {
+        if (forcing.isForceNone) done();
+      }
+    );
+    await api.tx.sudo.sudo(api.tx.staking.forceNoEras()).signAndSend(alice);
+  });
+
+  test('Force new era', async done => {
+    await api.query.staking.forceEra(
+      (forcing: Forcing) => {
+        if (forcing.isForceNew) done();
+      }
+    );
+    await api.tx.sudo.sudo(api.tx.staking.forceNewEra()).signAndSend(alice);
+  });
+
+  test('Force new era always', async done => {
+    await api.query.staking.forceEra(
+      (forcing: Forcing) => {
+        if (forcing.isForceAlways) done();
+      }
+    );
+    await api.tx.sudo.sudo(api.tx.staking.forceNewEraAlways()).signAndSend(alice);
+  });
+
+  test('Force unstake', async done => {
+    // Use charlie account as bob stash, it's simpler than funding a new account.
+    const bob_stash = keyring.addFromUri('//Charlie');
```

Comment:
I would suggest you keep calling this key: charlie_stash. If it is okay for charlie to sign and send bob's bond tx, then the whole test reads right as well and there would be no confusion over bob and bob_stash.

Reply:
ended up finding the correct funded uri for bob/stash